src/include/baselib/jni/JniEnvironment.h

```diff
@@ -136,6 +137,61 @@ namespace bl
                 return m_jniEnv -> GetVersion();
             }
 
+            jobjectRefType getObjectRefType( SAA_in const jobject object ) const NOEXCEPT
+            {
+                return m_jniEnv -> GetObjectRefType( object );
```

Comment:
Actually it is needed IMO... passing invalid object reference indicates a bug and / or process state corruption (in which case it is best to terminate immediately)... if the returned value is JNIInvalidRefType it should RIP terminate the process. It is probably best to leave the method as NOEXCEPT and throw an exception (UnexpectedException), but wrap it in BL_NOEXCEPT_* block which should cause RIP if an exception occurs.